lib/image-editor/class-image-editor-crop.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Start: Include for phase 2
+ * Image Editor: Image_Editor_Crop class
+ *
+ * @package gutenberg
+ * @since 7.x ?
+ */
+
+/**
+ * Crop image modifier.
+ */
+class Image_Editor_Crop extends Image_Editor_Modifier {
+	/**
+	 * Pixels from the left for the crop.
+	 *
+	 * @var integer
+	 */
+	private $crop_x = 0;
+
+	/**
+	 * Pixels from the top for the crop.
+	 *
+	 * @var integer
+	 */
+	private $crop_y = 0;
+
+	/**
+	 * Width in pixels for the crop.
+	 *
+	 * @var integer
+	 */
+	private $width = 0;
+
+	/**
+	 * Height in pixels for the crop.
+	 *
+	 * @var integer
+	 */
+	private $height = 0;
+
+	/**
+	 * Constructor.
+	 *
+	 * Will populate object properties from the provided arguments.
+	 *
+	 * @param integer $crop_x Pixels from the left for the crop.
+	 * @param integer $crop_y Pixels from the top for the crop.
+	 * @param integer $width  Width in pixels for the crop.
+	 * @param integer $height Height in pixels for the crop.
+	 */
+	public function __construct( $crop_x, $crop_y, $width, $height ) {
+		$this->crop_x = floatval( $crop_x );
```

Comment:
Why are we using `floatval` is the types are documented as `integer`? Which is correct?